src/extension.ts

```diff
@@ -59,7 +59,13 @@ export async function activate(context: ExtensionContext) {
     context.subscriptions.push(languages.registerDefinitionProvider('http', new CustomVariableDefinitionProvider()));
     context.subscriptions.push(languages.registerReferenceProvider('http', new CustomVariableReferenceProvider()));
     context.subscriptions.push(languages.registerDocumentSymbolProvider('http', new HttpDocumentSymbolProvider()));
-    new VariableDiagnosticsProvider().activate(context.subscriptions);
+
+    const diagnosticsProviders = new VariableDiagnosticsProvider();
+    workspace.onDidOpenTextDocument(diagnosticsProviders.checkVariables, this, context.subscriptions);

```

Comment:
It seems the `this` target is not right, we should use `diagnosticsProviders` instead.

Reply:
Will fix